group-generators/helpers/data-providers/talentlayer/index.ts

```diff
@@ -30,14 +30,37 @@ export class TalentLayerProvider extends GraphQLProvider {
   }
 
   /**
-   * Get all users with a talent layer id
+   * Get all users with a TalentLayer id
    */
   private async processUsersWithTalentLayerId(): Promise<FetchedData> {
+    const BATCH_SIZE = 10;
+    const LIMIT = 1000;
+
     const dataProfiles: FetchedData = {};
-    const response: Users = await getUsersWithTalentLayerIdQuery(this);
-    response.users.forEach((user) => {
-      dataProfiles[user.address] = 1;
-    });
+    let continueFetch = true;
+    let countBatch = 0;
+
+    while (continueFetch) {
+      const requests = [];
+      for (
+        let i = BATCH_SIZE * countBatch;
+        i < BATCH_SIZE * (countBatch + 1);
+        i++
+      ) {
+        requests.push(getUsersWithTalentLayerIdQuery(this, i * LIMIT));
+      }
+      const responses: Users[] = await Promise.all(requests);
```

Comment:
It would be better if we add some error management.
For example here you can do a Promise.all().then().catch()

But for others parts of the code when Promise is not used let's use try catch 👍

Reply:
so I used you utils function retryRequest that catch error on every API async call. Is that ok ?